### ISSUE TITLE
Add state boundaries to overlap and delta maps

### ISSUE THREAD

**Issue by jcoliver** [Big-Biodiversity-Collaborative/SwallowtailClimateChange#88]
Add boundaries of states (CA, MZ, US) to maps, maybe lighter or thinner than country boundaries.

**Comment by ezylstra** [comment]
Ok @jcoliver. Boy did I get caught in the ggplot/raster circle of hell today.  The good news is that I've amended the `overlap_map()` function to include state/province boundaries in 4c3e0adf. There are two things left to do:
- similarly update code to create delta maps
- figure out why the projection area (for just current time period?) extends south of Mexico, and prevent that from happening. 

Also, I realized when looking at maps for some species with large geographic ranges (eg, P. machaon), that we probably want to use a different projection for maps because things start looking crazy.  I think I've worked out most of the kinks, but the kicker is that the `terra::project()` function is REALLY slow for large rasters. My current thought is that we add an argument to the map functions that would indicate when we want to use the Equal Area Albers projection, but if you have other ideas, I'd be happy to hear them.  Right now I haven't added this functionality to the repo, but can do that when we're back in Jan.
